cognito-post-authentication/src/main/java/no/unit/nva/cognito/service/UserApiClient.java

```diff
@@ -59,17 +60,33 @@ public UserApiClient(HttpClient httpClient, ObjectMapper objectMapper, Environme
 
     @Override
     @JacocoGenerated
-    public void createUser(User user) {
+    public Optional<User> createUser(User user) {
+        logger.info("Requesting user creation for username: " + user.getUsername());
+        return createNewUser(user)
+            .stream()
+            .filter(responseIsSuccessful())
```

Comment:
refactor responseIsSuccessful to the following and call the filter like `filter(this::responseIsSuccessful)`

```
 private boolean responseIsSuccessful(HttpResponse<String> response) {
        return response.statusCode() == HttpStatus.SC_OK;
    }
```